local/owr_video_renderer.c

```diff
@@ -375,10 +382,60 @@ static GstElement *owr_video_renderer_get_element(OwrMediaRenderer *renderer, gu
     return renderer_bin;
 }
 
+/**
+ * owr_video_renderer_set_request_context_callback: Configure the GClosure to
+ * invoke when a GL context is required by the GStreamer pipeline running the
+ * renderer.
+ *
+ * @renderer:
+ * @callback: function providing the GstContext as return value. The parameters passed are a const gchar* representing the context type and the @user_data gpointer.
+ * @user_data: extra data passed as first argument of the @callback.
+ * @destroy_data: function invoked when disposing the user_data of the GClosure.
+ */
+void owr_video_renderer_set_request_context_callback(OwrVideoRenderer *renderer, OwrVideoRendererRequestContextCallback callback, gpointer user_data, GDestroyNotify destroy_data)
+{
+    if (renderer->priv->request_context)
+        g_closure_unref(renderer->priv->request_context);
+
+    renderer->priv->request_context = g_cclosure_new(G_CALLBACK(callback), user_data, (GClosureNotify) destroy_data);
+    g_closure_set_marshal(renderer->priv->request_context, g_cclosure_marshal_generic);
+}
+
+static GstBusSyncReply _owr_handle_gst_sync_message(GstBus *bus, GstMessage *message, gpointer user_data)
+{
+    OwrVideoRenderer *renderer = OWR_VIDEO_RENDERER(user_data);
+
+    OWR_UNUSED(bus);
+
+    if (renderer->priv->request_context == NULL)
+        return GST_BUS_PASS;
+
+    if (GST_MESSAGE_TYPE(message) == GST_MESSAGE_NEED_CONTEXT) {
+        GstContext* context;
+        const gchar *context_type;
+        GValue result = G_VALUE_INIT;
+        GValue params[1] = { G_VALUE_INIT };
+
+        gst_message_parse_context_type(message, &context_type);
+        g_value_init(&result, G_TYPE_POINTER);
+        g_value_init(&params[0], G_TYPE_STRING);
+        g_value_set_string(&params[0], context_type);
+        g_closure_invoke(renderer->priv->request_context, &result, 1, (const GValue*) &params, NULL);
```

Comment:
Note that the closure usage here is not necessarily thread-safe, depending on what the API requiremements of the setter are. Someone might change the closure while it is being invoked here from some random GStreamer thread